tests/queries/0_stateless/03644_min_level_for_wide_part.sql

```diff
@@ -0,0 +1,25 @@
+DROP TABLE IF EXISTS t_03644_min_level_for_wide_part;
+
+-- Can produce initial parts with level 1
+SET optimize_on_insert = 0;
+
+CREATE TABLE t_03644_min_level_for_wide_part (x int) ENGINE = MergeTree ORDER BY x
+SETTINGS min_bytes_for_wide_part = 0, min_rows_for_wide_part = 0, min_level_for_wide_part = 2;
+
+INSERT INTO t_03644_min_level_for_wide_part VALUES (1);
+
+SELECT level, part_type FROM system.parts WHERE active AND  table = 't_03644_min_level_for_wide_part';
+
+INSERT INTO t_03644_min_level_for_wide_part VALUES (2);
+
+OPTIMIZE TABLE t_03644_min_level_for_wide_part FINAL;
```

Comment:
Let's set `alter_sync = 2` to avoid flaky tests in case of slow builds (like sanitizers, debug, etc).

Reply:
Done